src/hera/task.py

```diff
@@ -272,6 +276,7 @@ def __init__(
         self.is_exit_task: bool = False
         self.depends: Optional[str] = None
         self.when: Optional[str] = None
+        self.ports = ports or []
```

Comment:
I think this should stay as `ports` so that it takes a `None` value when one is not provided. That way when line 935 builds the ports it would not create an empty list. I believe that, currently, the empty list of ports would show up in the YAML manifest submitted to Argo. Then line 935 becomes: 
```python
ports=None if self.ports is None else [p.build() for p in self.ports]
```

Reply:
Aha, good catch, so just to make sure:
line 279: self.port = ports
line 935: ports=None if self.ports is None else [p.build() for p in self.ports]